x-pack/plugins/observability_solution/observability_ai_assistant_app/public/functions/visualize_esql.tsx

```diff
@@ -149,11 +156,20 @@ export function VisualizeESQL({
     []
   );
 
+  const dataView = useMemo(() => {
+    if (dataViewAsync.value) {
+      if (dataViewAsync.value.fields.getByName('@timestamp')?.type === 'date') {
+        dataViewAsync.value.timeFieldName = '@timestamp';
+      }
+      return dataViewAsync.value;
+    }
+  }, [dataViewAsync.value]);
```

Comment:
no need to memoize this I think, and this is probably better off in the callback for dataViewAsync (e.g. you can use `getESQLAdHocDataView().then( dataView => { /*timestamp stuff*/ ; return dataView; })`), just as easy and no side effects for useMemo

Reply:
Done https://github.com/elastic/kibana/pull/184106/commits/469de5e05cfd95faac1678db733d97f4ddbde113